R/digital-ocean.R

```diff
@@ -1,470 +1,42 @@
-  } else {
-    # Append semicolon if necessary
-    if (!grepl(";\\s*$", preflight)){
-      preflight <- paste0(preflight, ";")
-    }
-  }
-  service <- gsub("\\$PREFLIGHT\\$", preflight, service)
-
-  if (missing(swagger)){
-    swagger <- "FALSE"
-  } else {
-    swagger <- "TRUE"
-  }
-  service <- gsub("\\$SWAGGER\\$", swagger, service)
-
-  servicefile <- tempfile()
-  writeLines(service, servicefile)
-
-  remotePath <- file.path("/etc/systemd/system", paste0(serviceName, ".service"))
-
-  analogsea::droplet_upload(droplet, servicefile, remotePath)
-  analogsea::droplet_ssh(droplet, "systemctl daemon-reload")
-
-  # TODO: add this as a catch()
-  file.remove(servicefile)
-
-  # TODO: differentiate between new service (start) and existing service (restart)
-  analogsea::droplet_ssh(droplet, paste0("systemctl start ", serviceName, " && sleep 1")) #TODO: can systemctl listen for the port to come online so we don't have to guess at a sleep value?
-  analogsea::droplet_ssh(droplet, paste0("systemctl restart ", serviceName, " && sleep 1"))
-  analogsea::droplet_ssh(droplet, paste0("systemctl enable ", serviceName))
-  analogsea::droplet_ssh(droplet, paste0("systemctl status ", serviceName))
-
-  ### NGINX ###
-  # Prepare the nginx conf file
-  conf <- readLines(system.file("server", "plumber-api.conf", package="plumber"))
-  conf <- gsub("\\$PORT\\$", port, conf)
-  conf <- gsub("\\$PATH\\$", path, conf)
-
-  conffile <- tempfile()
-  writeLines(conf, conffile)
-
-  remotePath <- file.path("/etc/nginx/sites-available/plumber-apis", paste0(path, ".conf"))
-
-  analogsea::droplet_upload(droplet, conffile, remotePath)
-
-  # TODO: add this as a catch()
-  file.remove(conffile)
-
-  if (forward){
-    do_forward(droplet, path)
-  }
-
-  analogsea::droplet_ssh(droplet, "systemctl reload nginx")
+#' @rdname digitalocean
+do_deploy_api <- function(...) {
+  lifecycle::deprecate_stop("1.0.0", "do_deploy_api()", "plumberDeploy::do_deploy_api()")
 }
 
-#' Forward Root Requests to an API
-#'
-#' @param droplet The droplet on which to act. It's expected that this droplet
-#'   was provisioned using [do_provision()].
-#' @param path The path to which root requests should be forwarded
 #' @export
-do_forward <- function(droplet, path){
-  # Trim off any leading slashes
-  path <- sub("^/+", "", path)
-  # Trim off any trailing slashes if any exist.
-  path <- sub("/+$", "", path)
-
-  if (grepl("/", path)){
-    stop("Can't deploy to nested paths. '", path, "' should not have a / in it.")
-  }
-
-  forward <- readLines(system.file("server", "forward.conf", package="plumber"))
-  forward <- gsub("\\$PATH\\$", paste0(path), forward)
-
-  forwardfile <- tempfile()
-  writeLines(forward, forwardfile)
-
-  analogsea::droplet_upload(droplet, forwardfile, "/etc/nginx/sites-available/plumber-apis/_forward.conf")
-
-  # TODO: add this as a catch()
-  file.remove(forwardfile)
-
-  invisible(droplet)
+#' @rdname digitalocean
+do_forward <- function(...) {
```

Comment:
We should be nicer about these deprecations. Specifically, I'm worried about the `library(plumberDeploy); library(plumber)` case